File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    nums = []
    for i in range(1, yellow + 1):
        if yellow % i == 0:
            nums.append(i)
            
    if len(nums) % 2 == 1:
        nums.insert(len(nums)//2, nums[len(nums)//2])
    
    for i in range(0, len(nums)//2):
        x = nums[i] + 2
        y = nums[len(nums) - 1 - i] + 2
        if 2 * x + 2 * y - 4 == brown:
            answer.append(y)
            answer.append(x)
            break
    

    
    
    return answer

File: 프로그래머스/2/42842. 카펫/카펫.java
import java.util.*;
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        
        for (int x = 1; x <= Math.sqrt(yellow); x++) {
            if (yellow % x != 0) continue;
            int y = yellow / x;
            if ((x + 2) * (y + 2) == brown + yellow) {
                answer[0] = Math.max(x, y) + 2;
                answer[1] = Math.min(x, y) + 2;
            }
        }
        
        return answer;
    }
}